examples/hackathon/critero_tsv.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+"""
+This file contains the data pipeline to read from a TSV file and output a DataFrame.
+"""
+import os
+
+import torcharrow.dtypes as dt
+from common import DEFAULT_CAT_NAMES, DEFAULT_INT_NAMES
+from torchdata.datapipes.iter import FileLister, FileOpener
+
+
+# Helper Functions
+def get_name(path_and_stream):
+    return os.path.basename(path_and_stream[0]), path_and_stream[1]
+
+
+# Convert TSV Strings to Float
+def str_to_num(s):
+    try:
+        return float(s)
+    except Exception:
+        return float("NaN")
+
+
+# Process each row
+def row_pre_process(row):
+    row = row[1:]  # Remove index column from TSV
+    for i in range(len(row)):
+        row[i] = str_to_num(row[i])
+    return tuple(row)
+
+
+DTYPE = dt.Struct(
+    [dt.Field("label", dt.float64)]
+    + [dt.Field(int_name, dt.Float64(nullable=True)) for int_name in DEFAULT_INT_NAMES]
+    + [dt.Field(cat_name, dt.Float64(nullable=True)) for cat_name in DEFAULT_CAT_NAMES]
+)
+
+source_dp = FileLister(".", masks="day_11_first_3k_rows.tsv")
+file_dp = FileOpener(source_dp, mode="b")
+map_dp = file_dp.map(get_name)
+csv_parser_dp = map_dp.parse_csv(delimiter="\t", skip_lines=1)
+columns_names = list(map_dp.parse_csv(delimiter="\t"))[0][1:]
+processed_dp = csv_parser_dp.map(row_pre_process)
+header_dp = processed_dp.header(1000)  # limit to 1000 rows to save time
+# TODO: The follow operation is very slow due to torcharrow.DataFrame
```

Comment:
The `.dataframe` operation is slow as the warning indicates.

cc: @wenleix

Reply:
Is the input list of tuple -- should be fixed via https://github.com/facebookresearch/torcharrow/pull/159

Reply:
Yes, the input is a list of tuple.
```python
print(list(header_dp)[0])  # This is a tuple (0.0, 118.0, ...)
```
Then in `.dataframe`, `batch` and `map` are called:
```python
batch_dp = source_dp.batch(dataframe_size)
# I just confirmed that the line of code below is the one that is very slow
df_dp = batch_dp.map(partial(torcharrow.DataFrame, dtype=dtype, columns=columns, device=device))  
return df_dp
```

Reply:
@NivekT Can you try TorchArrow with https://github.com/facebookresearch/torcharrow/pull/159 ? (BTW did you build TorchArrow locally ? :) )

Reply:
I just updated TorchArrow and rebuilt. I can confirm that it is 100x faster. Thanks!